### PR TITLE
Add code coverage to CI

### PR DESCRIPTION
**Pull request by sehrope**
Adds some code coverage report generation to Travis. This will create it as part of CI but I think you need to authorize / enable it to explicitly publish a comment to the PR with a report: https://docs.codecov.io/docs/pull-request-comments

Here's a sample of what get's generated for this PR: https://codecov.io/gh/sehrope/node-pg-types/tree/f502355098394ae3e159a28552c6c6caa2680eba

Existing coverage is pretty good (nice!) and most of what's missing is things we've already identified (ex: binary arrays).

Also, noticed that the travis config had an extra env var, PGUSER, that wasn't being referenced anywhere. It's been there since first adding that file so probably a part of wherever it was copied from (I'm guessing pg).

If you want to run coverage locally you do it via:

    npx nyc --reporter=html npm test && npx open-cli coverage/index.html

### CODE REVIEW

**Comment by sehrope** [comment]
Okay I guess the comments are already enabled :+1:

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,8 @@ language: node_js
...
   - '4'
   - 'lts/*'
-  - 'node'
-env:
-  - PGUSER=postgres
+matrix:
+  include:
+    - name: 'Code Coverage'
+      node_js: 'stable'
+      script: 'npx nyc --reporter=lcov npm test && npx codecov'
```
Comment (bendrucker) [inline .travis.yml]: Can you add these as devDependencies and then provide an `npm coverage` script rather than using `npx` here?

**Comment by sehrope** [comment]
We've used this module extensively through pg for years so happy to help.

File: .travis.yml (hunk 1 above)
Comment (sehrope) [reply .travis.yml]: Sure for including the dev dependencies but how about leaving the command itself in travis.yml (without `npx`)?I don't see a point to including the `codecov` execution in an npm script as nobody would run it locally outside a CI environment.

Instead we add a `npm run coverage` script that runs coverage and opens your browser to view the results: `nyc --reporter=html npm test && open-cli coverage/index.html`

Comment (bendrucker) [reply .travis.yml]: Agreed—I'd leave out npx here. A `coverage` and `coverage-ci` npm script would be nice.

**Comment by sehrope** [comment]
Rebased atop master, added the coverage items as dev dependencies, added `coverage` and `coverage-ci` npm scripts, and also added one more commit to cleanup the `.PHONY` line in the Makefile as it had a typo with a non-existent target (needed to update it anyway to add "coverage" to it).

**Comment by sehrope** [comment]
:tada:

File: package.json [hunk 2]
```diff
@@ -24,7 +26,10 @@
   },
   "homepage": "https://github.com/brianc/node-pg-types",
   "devDependencies": {
+    "codecov": "^3.5.0",
     "if-node-version": "^1.1.1",
+    "nyc": "^14.1.1",
+    "open-cli": "^5.0.0",
```
Comment (charmander) [inline package.json]: :/ https://github.com/sindresorhus/open-cli/blob/46ad34a9a4a19bab1cdba4ee71221e29b9d4aff1/cli.js#L3-L7

Comment (sehrope) [reply package.json]: Yes I wish it didn't include things like that and only supported opening existing files. I put it for convenience as you usually want to view the HTML report after running coverage but could leave it out and have it be done manually.

**Pull request merged**